File: src/main/java/ConsumeEvent.java
import org.jetbrains.annotations.NotNull;
import org.jnativehook.GlobalScreen;
import org.jnativehook.NativeHookException;
import org.jnativehook.NativeInputEvent;
import org.jnativehook.keyboard.NativeKeyEvent;
import org.jnativehook.keyboard.NativeKeyListener;

import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.AbstractExecutorService;
import java.util.concurrent.TimeUnit;
import java.util.logging.Level;
import java.util.logging.Logger;

public class ConsumeEvent implements NativeKeyListener {
    public ConsumeEvent() {
        // Create custom logger and level.
        Logger logger = Logger.getLogger(GlobalScreen.class.getPackage().getName());
        logger.setLevel(Level.WARNING);

        GlobalScreen.setEventDispatcher(new VoidDispatchService());
        try {
            GlobalScreen.registerNativeHook();
        } catch (NativeHookException e) {
//            e.printStackTrace();
        }

        GlobalScreen.addNativeKeyListener(this);
    }

    private static class VoidDispatchService extends AbstractExecutorService {
        private boolean running = false;

        public VoidDispatchService() {
            running = true;
        }

        public void shutdown() {
            running = false;
        }

        @NotNull
        public List<Runnable> shutdownNow() {
            running = false;
            return new ArrayList<Runnable>(0);
        }

        public boolean isShutdown() {
            return !running;
        }

        public boolean isTerminated() {
            return !running;
        }

        public boolean awaitTermination(long timeout, @NotNull TimeUnit unit) {
            return true;
        }

        public void execute(Runnable r) {
            r.run();
        }
    }

    public void nativeKeyPressed(NativeKeyEvent e) {
        if (e.getKeyCode() == NativeKeyEvent.VC_A) {
            System.out.print("Attempting to consume B event...\t");
            try {
                Field f = NativeInputEvent.class.getDeclaredField("reserved");
                f.setAccessible(true);
                f.setShort(e, (short) 0x01);

                System.out.print("[ OK ]\n");
            }
            catch (Exception ex) {
                System.out.print("[ !! ]\n");
                ex.printStackTrace();
            }
        }
    }

    public void nativeKeyReleased(NativeKeyEvent e) {
        if (e.getKeyCode() == NativeKeyEvent.VC_A) {
            System.out.print("Attempting to consume B event...\t");
            try {
                Field f = NativeInputEvent.class.getDeclaredField("reserved");
                f.setAccessible(true);
                f.setShort(e, (short) 0x01);

                System.out.print("[ OK ]\n");
            }
            catch (Exception ex) {
                System.out.print("[ !! ]\n");
                ex.printStackTrace();
            }
        }
    }

    public void nativeKeyTyped(NativeKeyEvent e) { /* Unimplemented */ }

    public static void main(String [] args) {
        new ConsumeEvent();
    }
}

File: src/main/kotlin/helper/ConsumeEvent.kt
package helper

import org.jnativehook.NativeInputEvent
import org.jnativehook.keyboard.NativeKeyEvent
import org.jnativehook.keyboard.NativeKeyListener
import java.util.*
import java.util.concurrent.AbstractExecutorService
import java.util.concurrent.TimeUnit

class ConsumeEvent : NativeKeyListener {
    class VoidDispatchService : AbstractExecutorService() {
        private var running = false
        override fun shutdown() {
            running = false
        }

        override fun shutdownNow(): List<Runnable> {
            running = false
            return ArrayList(0)
        }

        override fun isShutdown(): Boolean {
            return !running
        }

        override fun isTerminated(): Boolean {
            return !running
        }

        @Throws(InterruptedException::class)
        override fun awaitTermination(timeout: Long, unit: TimeUnit): Boolean {
            return true
        }

        override fun execute(r: Runnable) {
            r.run()
        }

        init {
            running = true
        }
    }



    private var pressedListener : ((e:NativeKeyEvent) -> Boolean)? = null
    /**@return os에 이벤트를 전달할건지 여부 */
    fun setPressedListener(listener: (e:NativeKeyEvent) -> Boolean){
        this.pressedListener = listener
    }


    private var releasedListener : ((e:NativeKeyEvent) -> Boolean)? = null
    /**@return os에 이벤트를 전달할건지 여부 */
    fun setReleasedListener(listener: (e:NativeKeyEvent) -> Boolean){
        this.releasedListener = listener
    }


    override fun nativeKeyPressed(e: NativeKeyEvent) {
        pressedListener?.invoke(e)?.let {
            if(!it) {
                try {
                    val f = NativeInputEvent::class.java.getDeclaredField("reserved")
                    f.isAccessible = true
                    f.setShort(e, 0x01.toShort())
//                    print("[ OK ]\n")
                } catch (ex: Exception) {
//                    print("[ !! ]\n")
                    ex.printStackTrace()
                }
            }
        }
    }

    override fun nativeKeyReleased(e: NativeKeyEvent) {
        releasedListener?.invoke(e)?.let {
            if(!it) {
                try {
                    val f = NativeInputEvent::class.java.getDeclaredField("reserved")
                    f.isAccessible = true
                    f.setShort(e, 0x01.toShort())
//                    print("[ OK ]\n")
                } catch (ex: Exception) {
//                    print("[ !! ]\n")
                    ex.printStackTrace()
                }
            }
        }

    }

    override fun nativeKeyTyped(e: NativeKeyEvent) { /* Unimplemented */
    }
}